src/modules/Elsa.Workflows.Runtime/Middleware/Activities/BackgroundActivityInvokerMiddleware.cs

```diff
@@ -122,7 +122,7 @@ private static void CaptureOutputIfAny(ActivityExecutionContext context)
                 continue;
 
             var output = (Output?)outputDescriptor.ValueGetter(activity);
-            context.Set(output, outputEntry.Value);
+            context.Set(output, outputEntry.Value, outputDescriptor.Name);
```

Comment:
This fixes the output name always being set as "output" in the activity output register.